agent/engine/ordering_integ_test.go

```diff
@@ -380,6 +383,51 @@ func TestDependencyHealthyTimeout(t *testing.T) {
 	waitFinished(t, finished, orderingTimeout)
 }
 
+// TestContainerPullDependsOnContainerSpecificResource
+func TestContainerPullDependsOnContainerSpecificResource(t *testing.T) {
```

Comment:
how is this testing the new code?

Reply:
I should probably add another test on the failed case but this is just making sure dependency resolution is working when a container creation is depending on container specific resources of the same container